.github/workflows/tests.yml

```
@@ -112,6 +112,17 @@ jobs:
       fail-fast: false
     steps:
       - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
```

If we just use `"3.10"`, then we don't need the `cache`/`lru_cache` shim in `utils.py`. I don't think we need to use the Python version we're checking for this bit, do we?

```suggestion
          python-version: "3.10"

---

I thought the same, but *maybe* it has an effect for the external-dependency version?

---

Oof. That makes me realise that if we have any packages that declare a dependency on `numpy` or `pandas`, that might cause issues for users of Python 3.7, since the latest versions of `numpy` and `pandas` only support Python 3.8+.

Still, I feel _inclined_ to say "Let's keep things simple for now, and just use 3.10 unconditionally". If some user comes to us and complains that the stubs package doesn't work properly on 3.7 or whatever, we can always rethink this :)